### ISSUE TITLE
Synchronous Job Results are getting cleared after default time even after specifying result_ttl = -1

### ISSUE THREAD

**Issue by AkashKabra11** [rq/rq#1416]
Hi, 
I am trying to make synchronous jobs using rq. I am instantiating the queues like this: 

```
from rq import Queue
my_queue = Queue('sync', is_async=False, connection=Redis())
```

Now, when I create a job like this: 
```
job = my_queue.enqueue(func_name, args=(my_args,), result_ttl=-1)
```
**The Synchronous job results are not persisted after 500 seconds**. On the other hand, job results are persisted for asynchronous jobs. On a quick debug, I found that **the issue lies in this line**: 
https://github.com/rq/rq/blob/fc7940c77b69704ad6c059385fd530c850e5f0be/rq/queue.py#L373

`enqueue` function internally calls `enqueue_call` which internally calls `enqueue_job`. This `enqueue_job` function calls the `run_job` function whenever is_async is false. run_job function looks like this: 
```
    def run_job(self, job):
        job.perform()
        job.set_status(JobStatus.FINISHED)
        job.save(include_meta=False)
        job.cleanup(DEFAULT_RESULT_TTL).  <--------- ISSUE
        return 
```
It performs cleanup after DEFAULT_RESULT_TTL irrespective of what is the result_ttl passed. It should perform cleanup after `job.result_ttl interval`. This small issue is disrupting the entire functionality of Synchronous jobs. I request the contributors to please make this change ASAP.

**Comment by selwin** [comment]
Please open a PR for this :)

**Comment by lowercase00** [comment]
Fixed on #1510
